src/erlcloud_cloudwatch_logs.erl

```diff
@@ -437,6 +464,85 @@ log_stream_order_by(undefined) -> <<"LogStreamName">>;
 log_stream_order_by(log_stream_name) -> <<"LogStreamName">>;
 log_stream_order_by(last_event_time) -> <<"LastEventTime">>.
 
+%%------------------------------------------------------------------------------
+%% @doc
+%%
+%% GetQueryResults action
+%% https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_GetQueryResults.html
+%%
+%% ===Example===
+%%
+%%   Returns the results from the specified query.
+%%
+%% `
+%%   application:ensure_all_started(erlcloud).
+%%   {ok, Config} = erlcloud_aws:auto_config().
+%%   {ok, Results} = erlcloud_cloudwatch_logs:get_query_results("12ab3456-12ab-123a-789e-1234567890ab", [], Config).
+%% `
+%%
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec get_query_results(QueryId, Options) -> Results
+      when QueryId :: string(),
+           Options :: [{out, map}],
+           Results :: {ok, query_results()} | {error, erlcloud_aws:httpc_result_error()}.
+get_query_results(QueryId, Options) ->
+    get_query_results(QueryId, Options, default_config()).
+
+-spec get_query_results(QueryId, Options, Config) -> Results
+      when QueryId :: string(),
+           Options :: [{out, map}],
+           Config :: aws_config(),
+           Results :: {ok, query_results()} | {error, erlcloud_aws:httpc_result_error()}.
```

Comment:
Should this type spec be adjusted to account for the non-`query_results()` `{ok, ...}` form?

Reply:
You are right. I was expecting the `dialyzer` to have a say, here, but it didn't :-(